<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>fann_GUI</class>
 <widget class="QMainWindow" name="fann_GUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>859</width>
    <height>564</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>fann_GUI</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>main</string>
       </attribute>
       <widget class="QLabel" name="label_9">
        <property name="geometry">
         <rect>
          <x>350</x>
          <y>40</y>
          <width>55</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>XOR</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkBox_A">
        <property name="geometry">
         <rect>
          <x>110</x>
          <y>110</y>
          <width>83</width>
          <height>22</height>
         </rect>
        </property>
        <property name="text">
         <string>A</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkBoxB">
        <property name="geometry">
         <rect>
          <x>220</x>
          <y>110</y>
          <width>83</width>
          <height>22</height>
         </rect>
        </property>
        <property name="text">
         <string>B</string>
        </property>
       </widget>
       <widget class="QPushButton" name="b_OK">
        <property name="geometry">
         <rect>
          <x>530</x>
          <y>200</y>
          <width>80</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>run</string>
        </property>
       </widget>
       <widget class="QLabel" name="result">
        <property name="geometry">
         <rect>
          <x>120</x>
          <y>240</y>
          <width>55</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="label_10">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>47</width>
          <height>13</height>
         </rect>
        </property>
        <property name="text">
         <string>file: </string>
        </property>
       </widget>
       <widget class="QLineEdit" name="pathNet">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>10</y>
          <width>661</width>
          <height>21</height>
         </rect>
        </property>
       </widget>
       <widget class="QPushButton" name="b_loadNet">
        <property name="geometry">
         <rect>
          <x>750</x>
          <y>10</y>
          <width>80</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Load Net</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkBoxC">
        <property name="geometry">
         <rect>
          <x>340</x>
          <y>110</y>
          <width>72</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>C</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="checkBoxD">
        <property name="geometry">
         <rect>
          <x>430</x>
          <y>110</y>
          <width>72</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>D</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>log</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="1">
         <widget class="QPushButton" name="b_logClear">
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QTextBrowser" name="logBox">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>trening</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="1">
         <widget class="QPushButton" name="b_startTrain">
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="b_save_trainData">
          <property name="text">
           <string>Save data</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QTextBrowser" name="trainingData">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="readOnly">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QProgressBar" name="progressBar">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="opcje">
       <property name="styleSheet">
        <string notr="true">background-color: rgb(145, 145, 145);
</string>
       </property>
       <attribute name="title">
        <string>opcje</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="2" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>hidden neurons</string>
          </property>
         </widget>
        </item>
        <item row="3" column="7" colspan="2">
         <widget class="QComboBox" name="net_type">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <item>
           <property name="text">
            <string>FANN::TRAIN_INCREMENTAL</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>FANN::TRAIN_BATCH</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>FANN::TRAIN_RPROP</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>FANN::TRAIN_QUICKPROP</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSpinBox" name="max_iterations">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="maximum">
           <number>9000000</number>
          </property>
          <property name="value">
           <number>3000000</number>
          </property>
          <property name="displayIntegerBase">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">border-color: rgb(0, 255, 0);</string>
          </property>
          <property name="text">
           <string>num_input</string>
          </property>
         </widget>
        </item>
        <item row="1" column="7">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>iterations_between_reports</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QSpinBox" name="num_output">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="maximum">
           <number>1000000</number>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="0" column="8">
         <widget class="QSpinBox" name="num_hidden">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="maximum">
           <number>1000000</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSpinBox" name="numInput">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="frame">
           <bool>true</bool>
          </property>
          <property name="maximum">
           <number>100000</number>
          </property>
          <property name="value">
           <number>2</number>
          </property>
         </widget>
        </item>
        <item row="0" column="5">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>num_layers</string>
          </property>
         </widget>
        </item>
        <item row="0" column="6">
         <widget class="QSpinBox" name="num_layers">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>9</number>
          </property>
          <property name="value">
           <number>3</number>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>num_output</string>
          </property>
         </widget>
        </item>
        <item row="0" column="7">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>num_hidden</string>
          </property>
         </widget>
        </item>
        <item row="1" column="8">
         <widget class="QSpinBox" name="iterations_between_reports">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="maximum">
           <number>1000000</number>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>learning_rate</string>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QDoubleSpinBox" name="learning_rate">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="value">
           <double>0.900000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>max_iterations</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QPushButton" name="b_loadTrainingData">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>Load training data</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QPushButton" name="b_saveNetfile">
          <property name="text">
           <string>save net file</string>
          </property>
         </widget>
        </item>
        <item row="4" column="4">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="6">
         <widget class="QDoubleSpinBox" name="desired_error">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="singleStep">
           <double>0.000100000000000</double>
          </property>
          <property name="value">
           <double>0.000100000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="5">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>desired_error</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QSpinBox" name="l1">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QSpinBox" name="l2">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QSpinBox" name="l3">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="5">
         <widget class="QSpinBox" name="l4">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="6">
         <widget class="QSpinBox" name="l5">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="7">
         <widget class="QSpinBox" name="l6">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="8">
         <widget class="QSpinBox" name="l7">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>859</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
